### PR TITLE
Split memory validation

### PR DESCRIPTION
**Pull request by alan-baker**
Moves memory instruction validation that occurred in validate_id.cpp into a separate pass.

* Tests were not modified.
* Some minor disassembly changes
* Some minor formatting changes

### CODE REVIEW

File: source/CMakeLists.txt [hunk 1]
```diff
@@ -303,6 +303,7 @@ set(SPIRV_SOURCES
   ${CMAKE_CURRENT_SOURCE_DIR}/val/validate_layout.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/val/validate_literals.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/val/validate_logicals.cpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/val/validate_memory.cpp
```
Comment (dj2) [inline source/CMakeLists.txt]: Can you add this to BUILD.gn as well?

File: source/val/validate_memory.cpp [hunk 2]
```diff
@@ -0,0 +1,599 @@
...
+
+}  // namespace
+
+spv_result_t ValidateMemoryInstructions(ValidationState_t& _) {
+  for (auto& inst : _.ordered_instructions()) {
+    switch (inst.opcode()) {
+      case SpvOpVariable:
+        if (auto error = ValidateVariable(_, inst)) return error;
+        break;
+      case SpvOpImageTexelPointer:
```
Comment (dj2) [inline source/val/validate_memory.cpp]: Move this down to 582 to keep all the empty ones together?

File: source/CMakeLists.txt (hunk 1 above)
Comment (alan-baker) [reply source/CMakeLists.txt]: done

File: source/val/validate_memory.cpp (hunk 2 above)
Comment (alan-baker) [reply source/val/validate_memory.cpp]: done

**Comment by alan-baker** [comment]
Rebased, squashed and submitted as d49bedcaa68a907e2b201bc372d69256e50f2ab5

**Pull request closed without merging**